src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts

```diff
@@ -97,4 +112,39 @@ export class FormDateTimePickerComponent implements ControlValueAccessor {
     get currentTimeZone(): string {
         return Intl.DateTimeFormat().resolvedOptions().timeZone;
     }
+
+    /**
+     * validates the value from the input field, if it is a date value. In case it is invalid, isInvalidDate is set to true.
+     * An empty input field is considered valid, in this case the value is assigned null
+     * @param event the change event
+     */
+    validateAndUpdateField(event: Event) {
+        const val = (event.target as HTMLInputElement).value;
+        //value is empty string in case the input field has been cleared (this is handled in updateEmptyField)
+        if (val != '') {
+            //date is InvalidDate or an actual date
+            const date = new Date(val);
+            if (this.isValidDate(date)) {
+                this.isInvalidDate = false;
+                this.value = date;
+            } else {
+                this.isInvalidDate = true;
+            }
+            this._onChange(this.value);
+            this.valueChanged();
+        }
+    }
+
+    /**
+     * directly updates the value (and removes the invalidDate error message) if the input field has been cleared
+     * @param inputValue the string value from the input field
+     */
+    updateEmptyField(inputValue: string) {
+        if (inputValue == '') {
```

Comment:
Are you using `==` on purpose? Or should this be `===`?

Reply:
I'm not sure, would it make a difference? Shouldn't `==` and `===` behave the same if the two objects being compared have the same type?

Reply:
Technically yes, however, it makes sense, as a general rule, to always use the strongest operator when possible. So of someone tries to change the value it fails.

`inputValue == 0` would be equivalent, for example. `inputValue === 0` wouldn't be.

Reply:
Ok, makese sense. I changed it :)